File: test_archives/cpp_tests/batch4/test_function5.cpp
#include "../catch.hpp"
#include "../functions/function5.cpp"

TEST_CASE("function5 returns the correct result for test case 1", "[function5]") {
    std::string str1 = "test";
    std::string str2 = "string";
    int int1 = 5;
    int int2 = 10;
    float float1 = 5.5;
    float float2 = 2.2;
    bool bool1 = true;
    REQUIRE(function5(str1, str2, int1, int2, float1, float2, bool1) == "tg502.5True");
}

TEST_CASE("function5 returns the correct result for test case 2", "[function5]") {
    std::string str1 = "Hello";
    std::string str2 = "World";
    int int1 = 4;
    int int2 = 6;
    float float1 = 3.5;
    float float2 = 1.7;
    bool bool1 = false;
    REQUIRE(function5(str1,

File: test_archives/py_tests/batch4/test_function5.py
from functions.function5 import function5
import pytest

def test_function5_1():
    str1 = "test"
    str2 = "string"
    int1 = 5
    int2 = 10
    float1 = 5.5
    float2 = 2.2
    bool1 = True
    assert function5(str1, str2, int1, int2, float1, float2, bool1) == 'tg502.5True'

def test_function5_2():
    str1 = "Hello"
    str2 = "World"
    int1 = 4
    int2 = 6
    float1 = 3.5
    float2 = 1.7
    bool1 = False
    assert function5(str1, str2, int1, int2, float1, float2, bool1) == 'Hd242.058823529411765False'